server/src/main/java/org/elasticsearch/cluster/metadata/AliasMetadata.java

```diff
@@ -51,8 +52,11 @@
     @Nullable
     private final Boolean isHidden;
 
+    @Nullable
+    private final Boolean isDataStream;
```

Comment:
I don't think this field is needed? I also don't see that is currently is used in this pr.
In `MetadataCreateDataStreamService`, data stream aliases can be created without this information? The fact that a `DataStreamTemplate` is defined should be sufficient?